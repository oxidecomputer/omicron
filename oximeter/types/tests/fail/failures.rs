// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at https://mozilla.org/MPL/2.0/.

#[derive(oximeter::Target)]
pub struct WrongTargetFieldType {
    pub x: f32,
}

#[derive(oximeter::Metric)]
pub struct WrongMetricFieldType {
    pub x: f32,
    pub value: f64,
}

#[derive(oximeter::Metric)]
pub struct WrongMetricDataType {
    pub value: f32,
}

#[derive(oximeter::Target)]
pub enum CantUseEnum {
    A,
    B,
}

#[derive(oximeter::Target)]
pub struct CantUseTupleStruct(i64);

#[derive(oximeter::Metric)]
pub enum CantUseEnumMetric {
    A,
}

#[derive(oximeter::Target)]
pub struct CantUseTupleStructMetric(i64);

#[derive(oximeter::Metric)]
pub struct NeedValueField {
    pub x: i64,
}

fn main() { }
