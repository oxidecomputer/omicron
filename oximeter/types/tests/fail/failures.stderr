error: Fields must be one of type: String, IpAddr, i64, bool, Uuid
 --> $DIR/failures.rs:3:5
  |
3 |     pub x: f32,
  |     ^^^

error: Fields must be one of type: String, IpAddr, i64, bool, Uuid
 --> $DIR/failures.rs:8:5
  |
8 |     pub x: f32,
  |     ^^^

error: Must be one of the supported data types: bool, i64, f64, String, Bytes, Cumulative<i64>, Cumulative<f64>, Histogram<i64>, or Histogram<f64>
  --> $DIR/failures.rs:14:5
   |
14 |     pub value: f32,
   |     ^^^

error: Can only be derived for structs with named fields
  --> $DIR/failures.rs:18:1
   |
18 | pub enum CantUseEnum {
   | ^^^

error: Can only be derived for structs with named fields
  --> $DIR/failures.rs:24:1
   |
24 | pub struct CantUseTupleStruct(i64);
   | ^^^

error: expected `struct`
  --> $DIR/failures.rs:27:5
   |
27 | pub enum CantUseEnumMetric {
   |     ^^^^

error: Can only be derived for structs with named fields
  --> $DIR/failures.rs:32:1
   |
32 | pub struct CantUseTupleStructMetric(i64);
   | ^^^

error: Must contain a field named "value"
  --> $DIR/failures.rs:35:27
   |
35 |   pub struct NeedValueField {
   |  ___________________________^
36 | |     pub x: i64,
37 | | }
   | |_^
